src/Channel.php

```diff
@@ -9,24 +9,26 @@
 /**
  * Interface for sending messages between execution contexts, such as two coroutines or two processes.
  *
- * @template TReceive
+ * @template-covariant TReceive
  * @template TSend
+ *
+ * @extends \Traversable<int, TReceive>
  */
-interface Channel extends Closable
+interface Channel extends Closable, \Traversable
 {
     /**
      * @param Cancellation|null $cancellation Cancels waiting for the next value. Note the next value is not discarded
      * if the operation is cancelled, rather it will be returned from the next call to this method.
      *
-     * @return TReceive Data received.
+     * @return TReceive|null Data received or {@code null} if the channel has closed.
```

Comment:
We allow sending `null` now but also use it as termination signal?

Reply:
Oops, forgot to revert this annotation back. Fixed.